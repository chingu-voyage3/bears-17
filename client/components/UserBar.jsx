import React from 'react';
import PropTypes from 'prop-types';

const UserBar = props => (
  <div className="user">
    <img className="user__avatar" width="32px" src={props.avatar} alt="" />
    <span className="user__name">{props.name}</span>
  </div>
);

UserBar.defaultProps = {
  name: '',
};

UserBar.propTypes = {
  avatar: PropTypes.string.isRequired,
  name: PropTypes.string,
};

export default UserBar;
