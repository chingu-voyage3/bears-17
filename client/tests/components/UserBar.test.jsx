import React from 'react';
import renderer from 'react-test-renderer';

import UserBar from 'Components/UserBar';

describe('UserBar', () => {
  it('should match the snapshot', () => {
    const userbar = renderer.create(<UserBar name="John Doe" avatar="img.jpg" />);

    expect(userbar.toJSON()).toMatchSnapshot();
  });
});
