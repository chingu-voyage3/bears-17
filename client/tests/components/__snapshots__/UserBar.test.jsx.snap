// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBar should match the snapshot 1`] = `
<div
  className="user"
>
  <img
    alt=""
    className="user__avatar"
    src="img.jpg"
    width="32px"
  />
  <span
    className="user__name"
  >
    John Doe
  </span>
</div>
`;
